### PR TITLE
Use correct language code zh_CN for the Chinese documentation

### PR DESCRIPTION
**Pull request by petterreinholdtsen**
The 'cn' code is for the country (China), while the language code is 'zh'.
This bring the language code in line with the code used in src/po/.

The linuxcnc-doc-cn package is renamed to linuxcnc-doc-zh-cn, as _ is not
valid in package names.  The new package is set to conflict and replace
on the old package.

### CODE REVIEW

**Comment by SebKuzminsky** [comment]
(Funny, I've made nearly this exact change in a local branch, but I was holding it back so as not to complicate your po4a work.)

Our other translations are identified by their language code ([ISO 639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes)) without the country code suffix ([ISO 3166](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2)), so for example `es` for Spanish rather than `es-MX` or `es-BO`.  I think the Chinese translation should continue that pattern unless and until there's a compelling reason to specialize the translation into country-specific flavors.

File: debian/control.docs.in [hunk 1]
```diff
@@ -37,12 +37,14 @@ Description: controlador de movimiento para máquinas CNC y robots (Español).
...
 
-Package: linuxcnc-doc-cn
+Package: linuxcnc-doc-zh-cn
 Provides: linuxcnc-doc
 Architecture: all
 Depends: ${misc:Depends}
 Recommends: xdg-utils
 Suggests: pdf-viewer
+Conflicts: linuxcnc-doc-cn
+Replaces: linuxcnc-doc-cn
```
Comment (SebKuzminsky) [inline debian/control.docs.in:47]: Good!

**Comment by SebKuzminsky** [comment]
The .desktop and .svg files need to have their contents updated.

**Comment by SebKuzminsky** [comment]
Here's my branch, in case there's anything useful for you there: https://github.com/LinuxCNC/linuxcnc/pull/new/rename-chinese-docs-iso-639-1

**Comment by petterreinholdtsen** [comment]
[Sebastian Kuzminsky]
> Our other translations are identified by their language code ([ISO
> 639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes)) without
> the country code suffix ([ISO
> 3166](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2)), so for
> example `es` for Spanish rather than `es-MX` or `es-BO`.  I think the
> Chinese translation should continue that pattern unless and until
> there's a compelling reason to specialize the translation into
> country-specific flavors.

The reason is in src/po/, where both zh_CN.po, po/zh_HK.po and
po/zh_TW.po is located.  I assume we might end up with documentation for
the same languages, thus went with the same language codes, as I know
the chinese in these three areas is quite different.

I'll update my branch with the fixes you proposed.

-- 
Happy hacking
Petter Reinholdtsen

**Comment by petterreinholdtsen** [comment]
I believe the patch is complete, and tested that it will not cause a conflict for my translate-po4a branch.

**Comment by petterreinholdtsen** [comment]
I just noticed the chinese deb is missing its PDF, both in master and with this patch.  I guess a fix for that can wait until after the po4a migration.

**Comment by andypugh** [comment]
On Fri, 3 Dec 2021 at 21:52, petterreinholdtsen ***@***.***>
wrote:

> I just noticed the chinese deb is missing its PDF, both in master and with
> this patch. I guess a fix for that can wait until after the po4a migration.
>
This is probably due to the fact that at the moment we don't even try:
https://github.com/LinuxCNC/linuxcnc/blob/master/docs/src/Submakefile#L395

The current Chinese docs in the ISO / web pages were "hand built" on my
Buster config at home and manually uploaded.

If we stop trying to support Stretch (which I believe to be the plan) then
this is a trivial fix.

-- 
atp

**Comment by petterreinholdtsen** [comment]
I created 57c946feaf6f08230e3297d5294f81e159671b67 to detect if xetex is available and build Chinese PDF when it is present, and adjusted d/configure to add xetex doc dependencies for non-Stretch builds.

**Comment by SebKuzminsky** [comment]
Thanks @petterreinholdtsen !

**Pull request merged**